AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Subscribe/DataStoreObserveQueryOperation.swift

```diff
@@ -177,6 +181,7 @@ class ObserveQueryTaskRunner<M: Model>: InternalTaskRunner, InternalTaskAsyncThr
             }
             self.log.verbose("Start ObserveQuery")
             self.subscribeToItemChanges()
+            self.subscribeToModelSyncedEvent()
```

Comment:
this looks correct 
-  `startObserveQuery` gets called from the subscriber defined in `func subscribeToDataStoreState()` 
- `resetState` gets called from the subscriber defined in `func subscribeToDataStoreState()` 

startObserveQuery should estaablish the subscriptions that send snapshots, resetState should cancel those subscriptions to prevent more snapshots from sending